src/pytorch_lightning/utilities/meta.py

```diff
@@ -251,45 +255,47 @@ def _set_meta_device() -> None:
                 setattr(mod, subclass.__name__, meta_class)
             continue
 
-        class _IsinstanceMetaclass(type(subclass)):
+        class _IsinstanceMetaclass(type(subclass)):  # type: ignore[misc]
             def __instancecheck__(self, instance: Any) -> bool:
                 """Overrides the ``isinstance`` check on ``_MaterializerModule`` objects."""
                 return isinstance(instance, self.__bases__[0])
 
         # Create a class subclassing current `subclass` overriding its new method.
         # this will enable use to use `torch.distributed.nn.utils.init_meta` to create a `meta`
         # version of the current subclass module
-        class _MaterializerModule(subclass, metaclass=_IsinstanceMetaclass):
+        class _MaterializerModule(subclass, metaclass=_IsinstanceMetaclass):  # type: ignore[valid-type, misc]
             @classmethod
             @contextmanager
-            def instantiation_context(cls):
+            def instantiation_context(cls) -> Generator[None, None, None]:
                 _unset_meta_device(from_created=True)
                 yield
                 _set_meta_device_populated(from_created=True)
 
             @classmethod
-            def materialize(cls, materialize_fn: Callable):
+            def materialize(cls, materialize_fn: Callable) -> Type:
                 with cls.instantiation_context():
                     obj = materialize_fn()
                 return obj
 
             @staticmethod
-            def add_subclasses(subclass):
+            def add_subclasses(subclass: Type) -> None:
                 """This is used to unroll the instantiation tree while creating the modules."""
                 # Don't store the LightningModule as skipped from the Meta process.
                 if subclass != pl.LightningModule:
                     __CREATED_MODULES__.add(subclass)
                 if subclass.__bases__[0] != torch.nn.modules.module.Module:
                     _MaterializerModule.add_subclasses(subclass.__bases__[0])
 
-            def __new__(cls, *args, **kwargs):
+            def __new__(cls, *args: Any, **kwargs: Any) -> Type:
                 subclass = cls.__bases__[0]
                 cls.add_subclasses(subclass)
                 with cls.instantiation_context():
                     obj = init_meta(subclass, *args, **kwargs)
+                    if isinstance(obj, Exception):
+                        raise obj
```

Comment:
Are you taking these changes from upstream?

Reply:
I wrote this line because `init_meta` function returns `MisconfigurationException` when `_TORCH_GREATER_EQUAL_1_10` is false.
If the error is returned, `obj.materialize` would be `none`. So I thought that it would be better to raise error explicitly